src/translations/indexMapper.js

```diff
@@ -0,0 +1,362 @@
+import { arrayFilter, arrayMap } from './../helpers/array';
+import { getListWithRemovedItems, getListWithInsertedItems } from './maps/utils/visuallyIndexed';
+import { rangeEach } from '../helpers/number';
+import IndexMap from './maps/indexMap';
+import SkipMap from './maps/skipMap';
+import MapCollection from './mapCollection';
+import localHooks from '../mixins/localHooks';
+import { mixin } from '../helpers/object';
+import { isDefined } from '../helpers/mixed';
+
+class IndexMapper {
+  constructor() {
+    /**
+     * Map storing the sequence of indexes.
+     *
+     * @type {IndexMap}
+     */
+    this.indexesSequence = new IndexMap();
+    /**
+     * Collection for different skip maps. Indexes marked as skipped in any map won't be rendered.
+     *
+     * @type {MapCollection}
+     */
+    this.skipCollection = new MapCollection();
+    /**
+     * Collection for another kind of mappings.
+     *
+     * @type {MapCollection}
+     */
+    this.variousMappingsCollection = new MapCollection();
```

Comment:
Inconsistency in naming: the property is called `*MappingCollection` but the class is called `MapCollection`. I propose to call the property `variousMapsCollection`.